src/npm.ts

```diff
@@ -282,6 +283,43 @@ export const refreshPackageJsonData = (
   }
 }
 
+export const refreshPnpmWorkspaceData = (
+  pnpmWorkspaceString: string,
+  pnpmWorkspaceFilePath: string,
+): Promise<void>[] => {
+  const cacheCutoff = new Date(new Date().getTime() - 1000 * 60 * 120) // 120 minutes
+
+  try {
+    const dependencies = getPnpmWorkspaceDependencyInformation(pnpmWorkspaceString).flatMap(
+      (d) => d.deps,
+    )
+
+    const promises = dependencies
+      .map(({ dependencyName }) => {
+        const cache = npmCache[dependencyName]
+        if (
+          cache === undefined ||
+          cache.asyncstate === AsyncState.NotStarted ||
+          (cache.item !== undefined && cache.item.date.getTime() < cacheCutoff.getTime())
+        ) {
+          const packageJsonPath = pnpmWorkspaceFilePath.replace(
```

Comment:
Is this too risky? Is it safe enough to assume the `package.json` file is a sibling to the `pnpm-workspace.yaml`?

Reply:
I believe there always must be a sibling `package.json` as a sibling to the workspace file.

Reply:
No, there usually is one, but there does not _have_ to be one. And if all you are using it for is to manage shared dependencies and not as a build tool, there will be _no_ `package.json` (only in each package)